core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java

```diff
@@ -433,4 +440,46 @@ public final BuilderT defaultsMode(DefaultsMode defaultsMode) {
     public final void setDefaultsMode(DefaultsMode defaultsMode) {
         defaultsMode(defaultsMode);
     }
+
+    /**
+     * If the region is a FIPS pseudo region (contains "fips"), this method returns a pair of values, the left side being the
+     * region with the "fips" string removed, and the right being {@code true}. Otherwise, the region is returned
+     * unchanged, and the right will be empty.
+     */
+    private static Pair<Region, Optional<Boolean>> transformFipsPseudoRegionIfNecessary(Region region) {
+        String id = region.id();
+
+        // "fips-us-west-1"
+        if (id.startsWith(FIPS_PREFIX)) {
+            String prefixRemoved = id.substring(FIPS_PREFIX.length());
+            return Pair.of(Region.of(prefixRemoved), Optional.of(true));
+        }
+
+        // "us-west-1-fips"
+        if (id.endsWith(FIPS_SUFFIX)) {
+            int end = id.length() - FIPS_SUFFIX.length();
+            String suffixRemoved = id.substring(0, end);
+            return Pair.of(Region.of(suffixRemoved), Optional.of(true));
+        }
+
+        // "query-fips-us-west-2"
+        if (id.contains(FIPS_INFIX)) {
+            String infixRemoved = StringUtils.replaceOnce(id, FIPS_INFIX, "-");
+            return Pair.of(Region.of(infixRemoved), Optional.of(true));
+        }
+
+        // "rekognition.fips-us-west-2"
+        if (id.contains(FIPS_PREFIX)) {
+            String infixRemoved = StringUtils.replaceOnce(id, FIPS_PREFIX, "");
+            return Pair.of(Region.of(infixRemoved), Optional.of(true));
+        }
+
+        // "rekognition-fips.us-west-2"
+        if (id.contains(FIPS_SUFFIX)) {
+            String infixRemoved = StringUtils.replaceOnce(id, FIPS_SUFFIX, "");
+            return Pair.of(Region.of(infixRemoved), Optional.of(true));
+        }
```

Comment:
Since we apparently have to handle infix, can we just do...
```java
String[] find = { "fips-", "-fips" }; // Should probably make these static to avoid needing to allocate a new array each time.
String[] replace = { "", "" };
return StringUtils.replaceEach(id, find, replace);
```

It's slightly different, since it completely strips "fips-" and "-fips" regardless of how many times it occurs, but [that's what ruby seems to do](https://github.com/aws/aws-sdk-ruby/blob/45214f454e35ecc35b0e9d531ce7db7c4a7d81a3/gems/aws-sdk-core/lib/aws-sdk-core/plugins/regional_endpoint.rb#L65).

Reply:
Cool that works for me.